### DISCUSSION TITLE
Which version should the `getVersion` get? the one in Cargo.toml or the one in tauri.conf.json? and why?

### DISCUSSION THREAD

**Discussion by huangjj27** [tauri-apps/tauri#5021] [Q&A] [answered]
I use the [`getVersion`](https://tauri.app/v1/api/js/modules/app#getversion) api to get a version of my app, expecting the version in tauri.conf.json since I set it differently from the one in Cargo.toml, but finally get the version the version in Cargo.toml unexpectly.

**Comment by FabianLars** [comment, marked as answer]
It's supposed to get the tauri.conf.json file and fall back to Cargo.toml only if there is no version in tauri.conf

**Reply by huangjj27** [reply]
So if I put the version into some profile-specific config, for example `dev.tauri.conf.json` ，I will miss that version?

**Reply by FabianLars** [reply]
As long as you specified it with the command line flag `-c`/`--config` it _should_ be picked up too 🤔

**Reply by huangjj27** [reply]
It turns out that my `dev.tauri.conf.json` 's `version` is referring to the `package.json`, hence the version falls back to the `Cargo.toml`'s one.

**Reply by FabianLars** [reply]
Actually it's supposed to catch that behavior too 😅
